src/hipscat/io/write_metadata.py

```diff
@@ -24,6 +24,8 @@ class HipscatEncoder(json.JSONEncoder):
     def default(self, o):
         if isinstance(o, Path):
             return str(o)
+        if isinstance(o, (type, np.dtype, pd.core.dtypes.base.ExtensionDtype)):
```

Comment:
Can you add test coverage for this line?

e.g. adding another element to the `dictionary` in the `test_write_json_file` that has a previously-unsupported-type?

https://github.com/astronomy-commons/hipscat/blob/main/tests/hipscat/io/test_write_metadata.py#L15

Reply:
Docs failing is due to another unrelated issue (see https://github.com/astronomy-commons/hipscat/pull/275).

The unit tests themselves are running normally, it looks, but github is experiencing degradations in Actions right now... https://www.githubstatus.com/